packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -104,10 +119,24 @@ export function useGetRestrictionUiType(
     objectId: entityId,
   })
 
+  const { data: isExternalFileHandle, isLoading: isLoadingExternalFile } =
+    useGetIsExternalFileHandle(entityId, {
+      enabled: options.enabled,
+    })
+
   if (!restrictionInformation) {
     return undefined
   }
 
+  if (options.enabled) {
+    if (isLoadingExternalFile) {
+      return undefined
+    }
+    if (isExternalFileHandle) {
+      return RestrictionUiType.AccessibleExternalFileHandle
+    }
+  }
```

Comment:
Unnecessary check, if `enabled` is false, `isLoadingExternalFile` will always be false.

I think it's ok to have the `isExternalFileHandle` check here; if this is true, then we definitely have met all ARs and we do have `canDownload`.

```suggestion
  if (isLoadingExternalFile) {
    return undefined
  }
  
  if (isExternalFileHandle) {
    return RestrictionUiType.AccessibleExternalFileHandle
  }
```